rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/PointerBlock.scala

```diff
@@ -0,0 +1,43 @@
+package coop.rchain.rspace.nextgenrspace.history
+import coop.rchain.rspace.Blake2b256Hash
+import coop.rchain.rspace.nextgenrspace.history.History.codecTrie
+import scodec.Codec
+import scodec.bits.BitVector
+import scodec.codecs.{provide, vectorOfN}
+import coop.rchain.shared.AttemptOps._
+
+final case class PointerBlock private (toVector: Vector[Trie]) {
+  def updated(tuples: List[(Int, Trie)]): PointerBlock =
+    new PointerBlock(tuples.foldLeft(toVector) { (vec, curr) =>
+      vec.updated(curr._1, curr._2)
+    })
+
+  def countNonEmpty: Int = toVector.count(_ != EmptyTrie)
+}
+
+object PointerBlock {
+  val length = 256
+
+  def create(): PointerBlock = new PointerBlock(Vector.fill(length)(EmptyTrie))
+
+  def create(first: (Int, Trie)): PointerBlock =
+    PointerBlock.create().updated(List(first))
+
+  def create(first: (Int, Trie), second: (Int, Trie)): PointerBlock =
+    PointerBlock.create().updated(List(first, second))
+
+  def fromVector(vector: Vector[Trie]): PointerBlock = new PointerBlock(vector)
```

Comment:
this method should validate the keys of the vector to be in 0.255, and that none of the values is null